### ISSUE TITLE
How to generate .mat files?

### ISSUE THREAD

**Issue by LightRay2** [orlitany/DeepFunctionalMaps#3]
Hello!
I tried to reproduce .mat files for test models, but didn't succeded. 10 num bins + radius 9 are correct for SHOT? Radius 9 is applied as the percent of the area?

Best regards,
Michael.

**Comment by orlitany** [comment]
Hi,
I'll have to find the parameters i used. In the meanwhile, here's a link to the [data](https://drive.google.com/drive/folders/1Sm76v3znqQvWw3OUIa-kvzbqnMk1a7fF?usp=sharing)

**Comment by LightRay2** [comment]
For those who faced the same problem: .off file coordinates should be premultiplied by 100 before calculating SHOT.

A brief test runs on provided model: quite well for FAUST and SCAPE, but nearly random for TOSCA. Is it expected or possibly i'm doing something wrong (of course, no premultiplication for TOSCA, it's already in sentimeters)?

**Comment by orlitany** [comment]
the provided model was supposed to serve as a mock. I'll try to upload one
that has been trained longer. For Tosca (dogs) we used a pretrained model
(on faust) and refined it on several tosca pairs.

**Comment by GitHubforXiaoming** [comment]
Hello!
`part_evecs: eigenvectors on source shape`
`part_evecs_trans: transposed part_evecs with mass matrix correction`
`model_evecs: eigenvectors on target shape`
`model_evecs_trans: transposed model_evecs with mass matrix correction`
I want to obtain these value from other models. Can you explain more about it since these defines are not clear for me ?

Best regards.
Xiaoming.

**Comment by hellohawaii** [comment]
> 
> 
> Hello!
> I tried to reproduce .mat files for test models, but didn't succeded. 10 num bins + radius 9 are correct for SHOT? Radius 9 is applied as the percent of the area?
> 
> Best regards,
> Michael.

I am trying to compute the SHOT descriptors, too! May I ask what value I should assign to the key 'min_neighs' to get the same .mat file?

**Comment by orlitany** [comment]
Added a description of SHOT parameters and the calc_shot function to a Utils folder.

**Comment by Adamwgoh** [comment]
Hi, I noticed that the results calculated using the provided `calc_shot` matlab binary do not produce the same results at the ones provided. Because of that, I had not been able to reproduce the results of the sample mesh, let alone other FAUST models (using the given pre-trained model). Is it possible the sample data's SHOT is not calculated with the same way?

If it is, is there a way to provide the rest of the header and .cpp code necessary for us to compile one that is not tied to Matlab?

**Comment by rFalque** [comment]
@orlitany can you add the shot_descriptor.cpp  and shot_descriptor.h as well?

**Comment by orlitany** [comment]
calc_shot.cpp is in Utils...
alternatively you can write prof. Tombari as described on the project webpage: http://www.vision.deis.unibo.it/research/80-shot

**Comment by orlitany** [comment]
> Hi, I noticed that the results calculated using the provided `calc_shot` matlab binary do not produce the same results at the ones provided. Because of that, I had not been able to reproduce the results of the sample mesh, let alone other FAUST models (using the given pre-trained model). Is it possible the sample data's SHOT is not calculated with the same way?
> 
> If it is, is there a way to provide the rest of the header and .cpp code necessary for us to compile one that is not tied to Matlab?

perhaps it's a scale issue. I think others have commented on this already, try scaling the models by a factor of 100

**Comment by rFalque** [comment]
There are some missing files related to the includes of the calc_shot.cpp
#include "shot_descriptor.h"

I guess, the tools used are the same as the toolbox used there https://github.com/zorah/KernelMatching/blob/cb79b080abcb87a42ec01cac383751e342805e99/tools/shot/shot_descriptor.cpp

**Comment by orlitany** [comment]
> There are some missing files related to the includes of the calc_shot.cpp
> #include "shot_descriptor.h"
> 
> I guess, the tools used are the same as the toolbox used there https://github.com/zorah/KernelMatching/blob/cb79b080abcb87a42ec01cac383751e342805e99/tools/shot/shot_descriptor.cpp

Yes, we used the same tools i that work. Thanks for pointing this out!

**Comment by shixue-qing** [comment]
Hello, I am a student and I have read a paper you published in 2017 called Deep Function Mapping. I want to know what the.cpp file is used for. I want to run it and find that it has no main function and no function that needs to be called inside.Also, I would like to know how Figure 7 came out?I really need the code of Figure 7, and I hope you can send it to me.Thanks a million.

**Comment by shixue-qing** [comment]
Hello, I need your Shrec '16 data set, I hope you can send it to me, thank you very much.

**Comment by orlitany** [comment]
try to follow the code here: https://github.com/OshriHalimi/unsupervised_learning_of_dense_shape_correspondence
Oshri did a better job maintaining this repo and the difference between the methods is in the loss function so you could easily swap those.
